src/pool/srv_target.c

```diff
@@ -1242,18 +1231,27 @@ pool_tgt_disconnect_all(struct ds_pool *pool)
  * Stop a pool. Must be called on the system xstream. Release the ds_pool
  * object reference held by ds_pool_start. Only for mgmt and pool modules.
  */
-void
+int
 ds_pool_stop(uuid_t uuid)
 {
 	struct ds_pool *pool;
 
 	ds_pool_failed_remove(uuid);
 
-	ds_pool_lookup(uuid, &pool);
-	if (pool == NULL)
-		return;
-	D_ASSERT(!pool->sp_stopping);
+	ds_pool_lookup_internal(uuid, &pool);
+	if (pool == NULL) {
+		D_INFO(DF_UUID ": not found\n", DP_UUID(uuid));
+		return 0;
+	}
+	if (pool->sp_stopping) {
+		int rc = -DER_BUSY;
+
+		DL_WARN(rc, DF_UUID ": already stopping", DP_UUID(uuid));
+		ds_pool_put(pool);
+		return rc;
```

Comment:
For future releases, yes, I think we can improve the behavior, for instance, by letting the retried pool destroy commands wait for the in-progress pool destroy operation on the same pool; for 2.6, I think it's acceptable to only address the main problem (i.e., the retried pool destroy command succeeds, even though the pool hasn't really been destroyed). This is a first step in the right direction, AFAICS, not a workaround that we would throw away immediately.

Reply:
I am fine to resolve the main problem firstly. In the real user environment, the user may complain but acceptable that, because he/she can re-execute the `pool destroy` or query the space when `pool destroy` hits -DER_BUSY. But I am afraid that some CI tests may failed with -DER_BUSY if only check `pool destroy` result without checking system space.

Reply:
OK. How about returning `-DER_AGAIN` instead, so that the control plane will retry the pool destroy operation after a backoff (observed to be around 10s)? Then we kind of get the "wait" we want, although it's not the most clean way of "waiting". The UI will looks like (I injected a `ds_pool_child` leak):

```
[liw2@boro-66 ~]$ time dmg -o ~/daos_control.yml pool destroy -rf test-pool-0
^C
real    0m11.489s
user    0m0.069s
sys     0m0.022s

[liw2@boro-66 ~]$ time dmg -o ~/daos_control.yml pool destroy -rf test-pool-0
Pool-destroy command failed: client: code = 510 description = "the *control.PoolDestroyReq request timed out after 5m0s"
ERROR: dmg: client: code = 510 description = "the *control.PoolDestroyReq request timed out after 5m0s"
ERROR: dmg: client: code = 510 resolution = "retry the request or check server logs for more information"

real    5m0.065s
user    0m1.599s
sys     0m0.092s
```

Reply:
Yes, DER_AGAIN seems better for that.